### DISCUSSION TITLE
Pip install web3 doesn't work

### DISCUSSION THREAD

**Discussion by thibautlehmann** [smartcontractkit/full-blockchain-solidity-course-py#45] [Q&A] [answered]
Hi guys, i have a hard time to install web3 by using the command "pip install web3" as it is mentionned in the video at 3:53hours.
It starts to install well the first packages and then i have the following error :

ERROR: Command errored out with exit status 1:
     command: 'C:\Users\thiba\AppData\Local\Microsoft\WindowsApps\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\thiba\\AppData\\Local\\Temp\\pip-install-tzcwek0_\\bitarray_70307835814b441fbbf99ae1b3e3e460\\setup.py'"'"'; __file__='"'"'C:\\Users\\thiba\\AppData\\Local\\Temp\\pip-install-tzcwek0_\\bitarray_70307835814b441fbbf99ae1b3e3e460\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\thiba\AppData\Local\Temp\pip-record-8jd0bkac\install-record.txt' --single-version-externally-managed --user --prefix= --compile --install-headers 'C:\Users\thiba\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\LocalCache\local-packages\Python39\Include\bitarray'
         cwd: C:\Users\thiba\AppData\Local\Temp\pip-install-tzcwek0_\bitarray_70307835814b441fbbf99ae1b3e3e460\
    Complete output (13 lines):
    running install
    running build
    running build_py
    creating build
    creating build\lib.win-amd64-3.9
    creating build\lib.win-amd64-3.9\bitarray
    copying bitarray\test_bitarray.py -> build\lib.win-amd64-3.9\bitarray
    copying bitarray\test_util.py -> build\lib.win-amd64-3.9\bitarray
    copying bitarray\util.py -> build\lib.win-amd64-3.9\bitarray
    copying bitarray\__init__.py -> build\lib.win-amd64-3.9\bitarray
    running build_ext
    building 'bitarray._bitarray' extension
    error: Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools": https://visualstudio.microsoft.com/visual-cpp-build-tools/
    ----------------------------------------
ERROR: Command errored out with exit status 1: 'C:\Users\thiba\AppData\Local\Microsoft\WindowsApps\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\thiba\\AppData\\Local\\Temp\\pip-install-tzcwek0_\\bitarray_70307835814b441fbbf99ae1b3e3e460\\setup.py'"'"'; __file__='"'"'C:\\Users\\thiba\\AppData\\Local\\Temp\\pip-install-tzcwek0_\\bitarray_70307835814b441fbbf99ae1b3e3e460\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\thiba\AppData\Local\Temp\pip-record-8jd0bkac\install-record.txt' --single-version-externally-managed --user --prefix= --compile --install-headers 'C:\Users\thiba\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\LocalCache\local-packages\Python39\Include\bitarray' Check the logs for full command output.

I tried to install MCV C++ 14.0 by following the link but still the same.

**Comment by AlbertoSinigaglia** [comment, marked as answer]
Follow one of these guides:

1. [From Microsoft Forum](https://docs.microsoft.com/en-us/answers/questions/136595/error-microsoft-visual-c-140-or-greater-is-require.html)
2. [This YouTube Video](https://www.youtube.com/watch?v=rcI1_e38BWs)
2. [StackOverFlow](https://stackoverflow.com/questions/58287344/pip-install-web3-not-installing)

**Reply by PatrickAlphaC** [reply]
edited for clarity for other users, thanks @AlbertoSinigaglia !

**Comment by thedevildude** [comment]
I faced the same issue so I downloaded Visual Studio Installer and installed the C++ Development Kit and restarted my system.

**Comment by thibautlehmann** [comment]
Problem solved, thx guys !

**Reply by PatrickAlphaC** [reply]
Awesome! How did you solve it?

**Reply by Programproductions** [reply]
@AlbertoSinigaglia thanks for this I had the same problem on a previous laptop as well as this one. I knew from previous experience I had to find a download from Microsoft to fix it - but just couldn't remember what it was - you save me a lot of grief!! Thanks.

Number 2 - The YouTube video was easy to follow. Just download the Visual Studio Installer and installed the C++ Development Kit and restarted my system now works.
